diff --git a/libraries/freertos_plus/standard/freertos_plus_tcp/include/FreeRTOS_IP_Private.h b/libraries/freertos_plus/standard/freertos_plus_tcp/include/FreeRTOS_IP_Private.h
index fce5b3150..cd18f15da 100644
--- a/libraries/freertos_plus/standard/freertos_plus_tcp/include/FreeRTOS_IP_Private.h
+++ b/libraries/freertos_plus/standard/freertos_plus_tcp/include/FreeRTOS_IP_Private.h
@@ -658,7 +658,7 @@
         /* Before accessing any member of this structure, it should be confirmed */
         /* that the protocol corresponds with the type of structure */
 
-        union
+        struct
         {
             IPUDPSocket_t xUDP;
             #if ( ipconfigUSE_TCP == 1 )
